server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java

```diff
@@ -345,16 +351,18 @@ public NewIndexType build() {
     }
 
     private NewIndexType buildWithoutSubfields() {
-      Map<String, Object> hash = new TreeMap<>();
-      hash.putAll(ImmutableMap.of(
-        "type", getFieldType(),
-        INDEX, disableSearch ? INDEX_NOT_SEARCHABLE : INDEX_SEARCHABLE,
-        "norms", valueOf(!disableNorms),
-        "store", valueOf(store)));
+      ImmutableMap.Builder<String, String> hash = ImmutableMap.builder();
+      hash.put("type", getFieldType())
+        .put(INDEX, disableSearch ? INDEX_NOT_SEARCHABLE : INDEX_SEARCHABLE)
+        .put("norms", valueOf(!disableNorms))
+        .put("store", valueOf(store));
+      if (FIELD_TYPE_KEYWORD.equals(getFieldType())) {
+        hash.put("doc_values", disabledDocValues ? "false" : "true");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "false" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)